rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/internal_interpreter.kt

```diff
@@ -382,6 +382,22 @@ class InternalInterpreter(val systemInterface: SystemInterface) {
                     }
                     assign(statement.target, value.negate())
                 }
+                is SubStmt -> {
+                    val minuend = if (statement.factor1 == null) {
```

Comment:
Factor1, factor2, and target are not very clear names.
In my opinion we should determine the minuend and subtrahend when performing the translation from the parse tree to the ast.
At this point we should simply access them from the AST node, not performing this logic

Reply:
but minuend can be factor 1 if  factor 1 is not blank or result if factor 1 is blank.
it's possible to perform this logic translating from the parse tree to the ast?

Reply:
yes, it is possible. There we can write all logic we desire using Kotlin.
The idea is to "normalize" the AST, making our work as easy as possible in the interpreter.

The parse tree (i.e., the context classes) is complicated because it derives directly from what we parse. The AST can be simpler and we can perform the simplification during the translation from the parse tree to the AST.